File: features/authority/authority/Role.go
package authority

import (
	"github.com/hailongz/golang/db"
)

type Role struct {
	db.Object
	Name	string	`json:"name" name:"name" title:"角色名" length:"64" unique:"ASC"`
	Title	string	`json:"title" name:"title" title:"角色说明" length:"255"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他选项 JSON 叠加" length:"-1"`
}

func (O *Role) GetName() string {
	return "role"
}

func (O *Role) GetTitle() string {
	return "角色"
}

File: features/authority/ts/Role.ts
import { int64, int32 } from "./lib/less";


/**
 * 角色
 * @type db
 */
export class Role {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 角色名
     * @unique ASC
     * @length 64
     */
    name: string = ''

    /**
     * 角色说明
     * @length 255
     */
    title: string = ''

    /**
     * 其他选项 JSON 叠加
     * @length -1
     */
    options: any

}